### PR TITLE
Close stale PRs without signed CLA

### PR DESCRIPTION
**Pull request by erlend-aasland**
See python/core-workflow#93

### CODE REVIEW

**Comment by erlend-aasland** [comment]
FYI, we're "speed-testing" this PR over at https://github.com/hugovk/cpython/pulls

**Review by sumeshir26 — APPROVED** [review]

**Comment by erlend-aasland** [comment]
@hugovk, based on the [first cron run](https://github.com/hugovk/cpython/runs/4756099591?check_suite_focus=true), I suggest the following changes:

1. Based on the debug output, use `only-pr-labels: 'CLA signed'` instead of `exempt-pr-labels: 'CLA not signed'` for the first step. It is probably easier to read as well.
2. Name each step. Currently both of them confusingly appear as `Run actions/stale@v4`.

**Comment by erlend-aasland** [comment]
> Actually, I found the step with the "exempt rule" confusing to read compared to the step with the "only rule". It seems like the two rules are processed differently as well. For `only-pr-labels`, the action aborts almost immediately. For `exempt-pr-labels` it appears to abort much later in the process.

<details>
<summary>Debug output from the PR which was caught by the exempt-rule</summary>

```
[#17] Pull request #17
[#17] Found this pull request last updated at: 2022-01-09T16:25:57Z
[#17] The option only-labels (​https://github.com/actions/stale#only-labels​) was not specified
[#17] └── Continuing the process for this pull request
[#17] Days before pull request stale: 1
[#17] The pull request is not closed nor locked. Trying to remove the close label...
[#17] ├── The close-pr-label (​https://github.com/actions/stale#close-pr-label​) option was not set
[#17] └── Skipping the removal of the close label
[#17] This pull request hasn't a stale label
[#17] Skipping this pull request because it has an exempt label
```

</details>
<details>
<summary>Debug output from the PR which was caught by the only-rule</summary>

```
[#18] Pull request #18
[#18] Found this pull request last updated at: 2022-01-09T16:25:48Z
[#18] The option only-labels (​https://github.com/actions/stale#only-labels​) was specified to only process issues and pull requests with all those labels (1)
[#18] └── Skipping this pull request because it doesn't have all the required labels
```

</details>

I prefer the latter.

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -13,10 +13,23 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/stale@v4
+    - name: "Check PR's with 'CLA signed' label"
```
Comment (hugovk) [inline .github/workflows/stale.yml]: PRs not PR's ([ref](https://docs.github.com/en/issues/tracking-your-work-with-issues/assigning-issues-and-pull-requests-to-other-github-users))

```suggestion
    - name: "Check PRs with 'CLA signed' label"
```

File: .github/workflows/stale.yml [hunk 2]
```diff
@@ -13,10 +13,23 @@ jobs:
...
+      uses: actions/stale@v4
       with:
         repo-token: ${{ secrets.GITHUB_TOKEN }}
+        only-pr-labels: 'CLA signed'
         stale-pr-message: 'This PR is stale because it has been open for 30 days with no activity.'
         stale-pr-label: 'stale'
         days-before-stale: 30
         days-before-close: -1
+
+    - name: "Check PR's with 'CLA not signed' label"
```
Comment (hugovk) [inline .github/workflows/stale.yml]:
```suggestion
    - name: "Check PRs with 'CLA not signed' label"
```

File: .github/workflows/stale.yml (hunk 1 above)
Comment (erlend-aasland) [reply .github/workflows/stale.yml]: Ah, bad typing habit. Thanks.

Comment (erlend-aasland) [reply .github/workflows/stale.yml]: Oh, this calls for a Life of Brian reference:

https://www.youtube.com/watch?v=DdqXT9k-050

**Review by Mariatta — APPROVED** [review]

**Comment by erlend-aasland** [comment]
@Mariatta, thank you for reviewing! I've added the do-not-merge label until we've tested this change properly on our forks (see the last TODO item). I'll remove that label if everything is all fine after we've finished testing.

**Comment by erlend-aasland** [comment]
@hugovk: the logs look much better with the recent changes applied => https://github.com/hugovk/cpython/runs/4769591131?check_suite_focus=true

**Comment by erlend-aasland** [comment]
I've tested this change on my fork, modified to mark a PR stale after 1 day, and close after 1 day (no CLA). I've verified, as expected, that PRs that are marked stale, and that received comments after being marked stale, will _not_ be closed (cc. @ambv).

For test PR 20, which received comments after being marked stale, the actions bot correctly removed the stale label on the following run. For test PR 21, which did not receive comments after being marked stale, the actions bot closed the PR on the following run.

Here's the PRs:
- erlend-aasland/cpython#19
- erlend-aasland/cpython#20
- erlend-aasland/cpython#21

(Removing the do-not-merge label, @Mariatta.)

**Review by hugovk — APPROVED** [review]
Worked as expected on https://github.com/hugovk/cpython/pull/17 and https://github.com/hugovk/cpython/pull/18 👍

**Review by raghavthind2005 — APPROVED** [review]

**Comment by erlend-aasland** [comment]
@ambv Are ok with this PR as it stands?

**Comment by ambv** [comment]
@erlend-aasland yes, let's land it.

**Pull request merged**